spec/supabase_js_v2.yml

```diff
@@ -345,6 +345,26 @@ pages:
           ```js
             const { data, error } = supabase.auth.setSession(refresh_token)
           ```
+  auth.refreshSession():
+    title: 'refreshSession()'
+    $ref: '@supabase/gotrue-js.GoTrueClient.refreshSession'
+    notes: |
+      - This method will refresh the session whether the current one is expired or not.
+      - For server-side use, pass in the current session which includes a refresh token; as `getSession()` likely won't be able to retrieve it.
```

Comment:
This should not be recommended because it's unsafe. If both the client and server are using the same refresh token, the client won't be aware that the refresh token has been used on the server-side. When the client tries to use the refresh token, it will lead to the reuse detection kicking in and all descending refresh tokens will be revoked.

Reply:
For server-side use, you should pass in the session into a new supabaseClient in your api handler. For example:

```js
app.get('/example', (req, res) => {
  const supabase = createClient(process.env.SUPABASE_URL, process.env.SUPABASE_KEY, { 
    global: { 
      headers: {
        Authorization: req.header['Authorization'] // contains the access token to make authenticated user requests on the server side
      }
    }
  })
  
  const res = await supabase.auth.getUser() // this will work since the above sets the authorization header with the user's JWT
  ...
})
```

Reply:
> For server-side use, you should pass in the session into a new supabaseClient in your api handler. For example:
> 
> ```js
> app.get('/example', (req, res) => {
>   const supabase = createClient(process.env.SUPABASE_URL, process.env.SUPABASE_KEY, { 
>     global: { 
>       headers: {
>         Authorization: req.header['Authorization'] // contains the access token to make authenticated user requests on the server side
>       }
>     }
>   })
>   
>   const res = await supabase.auth.getUser() // this will work since the above sets the authorization header with the user's JWT
>   ...
> })
> ```

Thanks for reviewing!

yes, your example is how I'm using a supabase client on the the server-side.

In my specific use-case, the client-side supabase client has `autoRefreshToken` and `persistSession` set to false. So refreshing tokens with a server-side supabase client is my only option (I believe). When I do that with `refreshSession('jwt')`, I update browser cookies. So I just figured people should know how to refresh tokens with a server-side client.

With that explanation, should I still remove the wording or is there different wording you have in mind to explain what I've laid out?

Reply:
@kangmingtay If the wording can just be tweaked, maybe something like this?

```diff
- - For server-side use, pass in the current session which includes a refresh token; as `getSession()` likely won't be able to retrieve it.
+ - For use with a server-side Supabase client, pass in the current session which includes a refresh token.
+ - This method should only be used with a server-side Supabase client in the following circumstances:
+  - You are not using a client-side Supabase client in your app.
+  - Your client-side Supabase client has `autoRefreshToken` set to `false`.
```

Reply:
After thinking about this more, I should also say that I don't use the client-side client to make db queries. That all happens with my server-side client. But I can definitely see where someone could get themselves in trouble here.

I like the idea of providing guidance for proper server-side usage, because there are definitely gotchas. But maybe the official stance is to never use this method on the server-side.

Awaiting your thoughts.

Reply:
Hmm let's just remove that sentence completely for now, i'll do up a separate guide on how to use auth / supabase-js on the server-side and mention the few gotchas that devs need to look out for.

For now, i think it's sufficient to just describe this method as one that will forcefully refresh the session.